rxbindings/src/main/java/com/amplifyframework/rx/RxAdapters.java

```diff
@@ -100,4 +103,85 @@ private RxAdapters() {}
     interface VoidCompletionEmitter<E> {
         void emitTo(Action onComplete, Consumer<E> onError);
     }
+
+    /**
+     * Interface that defines a contract for callback-style operations to be
+     * mapped to reactive-style operation types that offer a more idiomatic way
+     * of using reactive programming with the Amplify framework.
+     * @param <T> The type that represents the result of a given operation.
+     * @param <P> The type that represents the progress of a given operation.
+     * @param <E> The type of exception of a given operation.
+     */
+    interface RxProgressAwareCallbackMapper<T, P, E> {
+        Cancelable emitTo(Consumer<P> onProgress, Consumer<T> onItem, Consumer<E> onError);
+    }
+
+    /**
+     * Interface that should be implemented by reactive-style operations
+     * wishing to return a {@link Single} as its result.
+     * @param <T> The type that represents the result of a given operation.
+     */
+    interface RxSingleOperation<T> extends Cancelable {
+        /**
+         * Maps the result of a callback-style operation to a {@link Single}.
+         * @return A {@link Single} that emits a result or an error.
+         */
+        Single<T> observeResult();
+    }
+
+    /**
+     * Interface that should be implemented by reactive-style operations
+     * that emit progress information during processing.
+     * @param <P> The type that represents the progress of a given operation.
+     */
+    interface RxProgressAwareOperation<P> {
```

Comment:
Since progress is currently only defined for _Storage_, should we move these new `RxProgressAwareSingle` and `RxProgressAwareOperation` out of `RxAdapters`, and into `RxStorageBinding.java`? (Well, the interface might need to go in `RxStorageCategoryBehavior.java`?)

Reply:
That makes sense. I was thinking we'd run across some progress aware API in one of the other categories, but now that I think about it, I can't think of an example.